### PR TITLE
feat(build) Enable more realistic capabilities of sample-service to better imitate real services.

### LINKED ISSUE #43: Make Sample Service a mock service that can be used to test
The service needs to be made into a real sample service - one that we can use for testing within our CI/CD development pipelines.

[US6866](https://rally1.rallydev.com/#/detail/userstory/399760409008?fdp=true): EdgeX DevOps: Make Sample-Service an actual sample service

### PR DESCRIPTION
**Pull request by bill-mahoney**
feat(build) Enable more realistic capabilities of sample-service to better imitate real services.

These capabilities are based off of the edgex device-random service.

Sample-service running healthily inside edgex:

image

sample-serviceResponding to random int8 REST endpoint:

image

Building in Jenkins with edgeXBuildGoApp:
image

See the sandbox link to see a successful build pipeline. 

Signed-off-by: Bill Mahoney <bill.mahoney@intel.com>

Closes #43 
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/edgexfoundry/ci-management/blob/master/.github/CONTRIBUTING.md
- [ ] Tests for the changes have been added (for bug fixes / features)
- [x] Added labels
## PR Type
What kind of change does this PR introduce?

- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Documentation content changes
- [ ] Other... Please describe:

## Sandbox Testing
Test Links :
https://jenkins.edgexfoundry.org/sandbox/job/Functional-Testing/job/edgex-sample-service/4/console

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -1 +1,51 @@
...
+ARG ALPINE_PKG_BASE="build-base git openssh-client"
+ARG ALPINE_PKG_EXTRA=""
+
+LABEL license='SPDX-License-Identifier: Apache-2.0' \
+  copyright='Copyright (c) 2020: Intel'
+
+RUN sed -e 's/dl-cdn[.]alpinelinux.org/nl.alpinelinux.org/g' -i~ /etc/apk/repositories
+RUN apk add --no-cache ${ALPINE_PKG_BASE} ${ALPINE_PKG_EXTRA}
+
+WORKDIR $GOPATH/src/github.com/edgexfoundry/sample-service
```
Comment (ernestojeda) [inline Dockerfile:29]: This WORKDIR can just be /sample-service. GOPATH based workdir is pre Go 1.11

File: Dockerfile [hunk 2]
```diff
@@ -1 +1,51 @@
...
+
+LABEL license='SPDX-License-Identifier: Apache-2.0' \
+  copyright='Copyright (c) 2020: Intel'
+
+ENV APP_PORT=49999
+EXPOSE $APP_PORT
+
+COPY --from=builder /go/src/github.com/edgexfoundry/sample-service/cmd /
+COPY --from=builder /go/src/github.com/edgexfoundry/sample-service/LICENSE /
+COPY --from=builder /go/src/github.com/edgexfoundry/sample-service/Attribution.txt /
```
Comment (ernestojeda) [inline Dockerfile:49]: If you change the WORKDIR above, you will also have to change these paths to be rooted at `/sample-service`

File: Jenkinsfile [hunk 3]
```diff
@@ -13,10 +13,8 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 //
-@Library("edgex-global-pipelines@experimental") _
 
 edgeXBuildGoApp (
     project: 'sample-service',
-    goVersion: '1.12',
-    buildExperimentalDockerImage: true
```
Comment (ernestojeda) [inline Jenkinsfile:21]: I think it is OK to leave this here. I will help us test in the future.

File: Jenkinsfile [hunk 4]
```diff
@@ -13,10 +13,8 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 //
-@Library("edgex-global-pipelines@experimental") _
```
Comment (ernestojeda) [inline Jenkinsfile:16]: Please leave this in here. Sample service is usually setup to test experimental edgex-global-pipleines functionality

File: docker-compose.yml [hunk 5]
```diff
@@ -0,0 +1,385 @@
+# /*******************************************************************************
```
Comment (ernestojeda) [inline docker-compose.yml:1]: Why is this file needed? I dont think we want to confuse people outside of EdgeX if they stumble upon this repo. Just the source code is fine.

Comment (bill-mahoney) [reply docker-compose.yml:1]: I'll remove it. It took me a bit to figure out how to properly start up edgex with the sample-service so this docker-compose helped me out. 

But the other device services don't include such a compose file so I agree with removing it.

File: README.md [hunk 6]
```diff
@@ -1,5 +1,34 @@
 # sample-service
-Sample service is a basic hello-world application that is used by DevOps for build automation / validation of Jenkins Pipelines.
+Sample service is a basic hello-world application that is used by the EdgeX DevOps working group for build automation / validation of Jenkins Pipelines.
```
Comment (ernestojeda) [inline README.md:2]: Can we add a note in here something like:

## Not for production use
Maybe link back to the edgex-examples repo as we wont actively be updating this codebase to match the latest and greatest.

File: Jenkinsfile [hunk 7]
```diff
@@ -17,6 +17,6 @@
 
 edgeXBuildGoApp (
     project: 'sample-service',
-    goVersion: '1.12',
+    goVersion: '1.13'
```
Comment (ernestojeda) [inline Jenkinsfile:20]: Syntax issue here...missing comma
